libsolidity/codegen/ArrayUtils.h

```diff
@@ -77,9 +73,10 @@ class ArrayUtils
 	/// Stack post:
 	void popStorageArrayElement(ArrayType const& _type) const;
 	/// Appends a loop that clears a sequence of storage slots of the given type (excluding end).
+	/// @param _canOverflow whether the storage is treated as circular when clearing.
 	/// Stack pre: end_ref start_ref
 	/// Stack post: end_ref
-	void clearStorageLoop(Type const* _type) const;
+	void clearStorageLoop(Type const* _type, bool _canOverflow) const;
```

Comment:
This is a bit off-topic, but reviewing this PR made me realize what the reason for the limitation from #15911 may be. For a type that takes up whole storage it is not possible to store its size in `u256`/`uint`. It would wrap around to zero. Similarly in the start-end representation we have here we'd end up with an empty range - an attempt to clear an array of this size would unexpectedly not run out of gas because it would not clear anything.